core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -201,20 +201,6 @@ private SqlQueryExecution(
                 tableExecuteContextManager.unregisterTableExecuteContextForQuery(stateMachine.getQueryId());
             });
 
-            // when the query finishes cache the final query info, and clear the reference to the output stage
-            AtomicReference<SqlQueryScheduler> queryScheduler = this.queryScheduler;
-            stateMachine.addStateChangeListener(state -> {
-                if (!state.isDone()) {
-                    return;
-                }
-
-                // query is now done, so abort any work that is still running
-                SqlQueryScheduler scheduler = queryScheduler.get();
```

Comment:
Do we have test for aborting? Can it be that this listener covers some flow that is not covered by `SqlQueryScheduler`?